File: app/src/main/java/com/notahelloworld/kotlin/nullable/UserRepository.java
package com.notahelloworld.kotlin.nullable;

public class UserRepository {

  public String getUserName(int userId) {
    User user = new UserDAO().getUser(userId);
    return user.name;
  }

  class UserDAO {
    User getUser(int userId) {
      //database.query(userId)
      return null;
    }
  }

  class User {
    String name;
  }

}

File: app/src/main/kotlin/com/notahelloworld/kotlin/nullable/UserRepository.kt
package com.notahelloworld.kotlin.nullable

class UserRepository {

  fun getUserName(userId: Int): String? {
    val user = UserDAO().getUser(userId)
    return user?.name
  }

  internal inner class UserDAO {
    fun getUser(userId: Int): User? {
      //database.query(userId)
      return null
    }
  }

  internal inner class User {
    var name: String? = null
  }

}
